File: Sample-ObjC/Connectors/AppsFlyerConnector.m
//
//  AppsFlyerConnector.m
//  ObjcSample
//
//  Created by artur on 07.05.2020.
//  Copyright © 2020 artur. All rights reserved.
//

#import "AppsFlyerConnector.h"
#import "ServicesInfo.h"

@import Appodeal;
@import AppsFlyerLib;


@interface AppsFlyerConnector() <AppsFlyerTrackerDelegate>

@end


@implementation AppsFlyerConnector

- (void)inititalise:(Completion)completion {
    // Configure AppsFlyer
    [AppsFlyerTracker.sharedTracker setAppsFlyerDevKey:ServicesInfo.sharedInfo.appsFlyerDevId];
    [AppsFlyerTracker.sharedTracker setAppleAppID:ServicesInfo.sharedInfo.appsFlyerAppId];
    [AppsFlyerTracker.sharedTracker setDelegate:self];
    // Set isDebug to true to see AppsFlyer debug logs
    [AppsFlyerTracker.sharedTracker setIsDebug:true];
    
    [NSNotificationCenter.defaultCenter addObserver:self
                                           selector:@selector(didLaunch)
                                               name:UIApplicationDidBecomeActiveNotification
                                             object:nil];
    
    // Set AppsFlyer tracker
    NSMutableDictionary <NSString *, NSString *> *extras = [NSMutableDictionary dictionaryWithCapacity:1];
    extras[kAPDAppsFlyerIdExtrasKey] = AppsFlyerTracker.sharedTracker.getAppsFlyerUID;
    [Appodeal setExtras:extras];
    
    // Notify completion
    dispatch_async(dispatch_get_main_queue(), ^{
        completion ? completion() : nil;
    });
}

- (void)didLaunch {
    [AppsFlyerTracker.sharedTracker trackAppLaunch];
}

- (void)dealloc {
    [NSNotificationCenter.defaultCenter removeObserver:self];
}

#pragma mark - AppsFlyerTrackerDelegate

- (void)onConversionDataFail:(NSError *)error {}

- (void)onConversionDataSuccess:(NSDictionary *)conversionInfo {
    [Appodeal setSegmentFilter:conversionInfo];
}

@end

File: Sample-Swift/AppsFlyerConnector.swift
//
//  AppsflyerConnector.swift
//  Sample
//
//  Created by Stas Kochkin on 04.05.2020.
//  Copyright © 2020 com.appodeal. All rights reserved.
//

import Foundation
import AppsFlyerLib
import Appodeal



final class AppsFlyerConnector: NSObject, Connector {
    func initialise(completion: @escaping Completion) {
        // Configure AppsFlyer
        AppsFlyerTracker.shared().appsFlyerDevKey = servicesInfo.appsFlyer.devKey
        AppsFlyerTracker.shared().appleAppID = servicesInfo.appsFlyer.appId
        AppsFlyerTracker.shared().delegate = self
        // Set isDebug to true to see AppsFlyer debug logs
        AppsFlyerTracker.shared().isDebug = true
        // Register notifications
        NotificationCenter.default.addObserver(
            self,
            selector: #selector(didLaunch),
            name: UIApplication.didBecomeActiveNotification,
            object: nil
        )
        // Get AppsFlyer UID
        Appodeal.setExtras([
            kAPDAppsFlyerIdExtrasKey: AppsFlyerTracker.shared().getAppsFlyerUID()
        ])
        DispatchQueue.main.async(execute: completion)
    }
    
    @objc private func didLaunch() {
        AppsFlyerTracker.shared().trackAppLaunch()
    }
}


extension AppsFlyerConnector: AppsFlyerTrackerDelegate {
    func onConversionDataSuccess(_ conversionInfo: [AnyHashable : Any]) {
        Appodeal.setSegmentFilter(conversionInfo)
    }
    
    func onConversionDataFail(_ error: Error) {}
}
